src/main/java/seedu/hirehub/logic/commands/StatusCommand.java

```diff
@@ -28,70 +21,72 @@ public class StatusCommand extends Command {
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
+        Application applicationToUpdate = findApplication(model);
 
-        if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        if (status.equals(applicationToUpdate.status)) {
+            throw new CommandException(String.format(MESSAGE_DUPLICATE_STATUS, status));
         }
 
-        Person personToUpdateStatus = lastShownList.get(index.getZeroBased());
-        Person statusUpdatedPerson = updatePersonStatus(personToUpdateStatus);
-
-        if (personToUpdateStatus.equals(statusUpdatedPerson) && model.hasPerson(statusUpdatedPerson)) {
-            throw new CommandException(String.format(MESSAGE_DUPLICATE_PERSON, status.toString()));
-        }
+        Application editedApplication = new Application(applicationToUpdate.getPerson(),
+                applicationToUpdate.getJob(), status);
 
-        model.setPerson(personToUpdateStatus, statusUpdatedPerson);
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-        return new CommandResult(String.format(MESSAGE_STATUS_PERSON_SUCCESS,
-                index.getOneBased(), status.toString()));
+        model.setApplication(applicationToUpdate, editedApplication);
+        model.updateFilteredApplicationList(PREDICATE_SHOW_ALL_APPLICATIONS);
+        return new CommandResult(String.format(MESSAGE_STATUS_PERSON_SUCCESS, status));
     }
 
     /**
-     * Updates a person with a new user-provided status among one of the five enum values
+     * Finds an application with the given job title and the email address of candidate
+     * @param model to retrieve the filtered application list
      */
-    public Person updatePersonStatus(Person personToUpdateStatus) {
-        assert personToUpdateStatus != null;
-
-        Name updatedName = personToUpdateStatus.getName();
-        Phone updatedPhone = personToUpdateStatus.getPhone();
-        Email updatedEmail = personToUpdateStatus.getEmail();
-        Country updatedCountry = personToUpdateStatus.getCountry();
-        Comment updatedComment = personToUpdateStatus.getComment();
-        Set<Tag> updatedTags = personToUpdateStatus.getTags();
-
-        return new Person(updatedName, updatedPhone, updatedEmail, updatedCountry, status,
-                updatedComment, updatedTags);
+    public Application findApplication(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Application> lastShownApplicationList = model.getFilteredApplicationList();
+        Job jobToFind = new Job(jobTitle, "", 10);
+
+        for (int i = 0; i < lastShownApplicationList.size(); i++) {
+            if (email.equals(lastShownApplicationList.get(i).getPerson().getEmail())
+                    && jobToFind.isSameJob(lastShownApplicationList.get(i).getJob())) {
+                return lastShownApplicationList.get(i);
```

Comment:
Would it be better to enumerate through items in the list instead of iterating through index?

Reply:
Can